app/ui/src/app/integrations/edit-page/step-configure/data-mapper/data-mapper-host.component.ts

```diff
@@ -108,31 +108,37 @@ export class DataMapperHostComponent extends FlowPage implements OnInit {
     // TODO not sure what to do for `none` or `any` here
     switch (kind) {
       case 'java':
-        this.addInitializationTask();
+
         const docDef: DocumentDefinition = this.cfg.addJavaDocument(
           type,
           isSource
         );
-        this.support.requestJavaInspection(connectorId, type).subscribe(
-          data => {
-            const inspection: string = data['_body'];
-            log.infoc(
-              () => 'Precomputed java document found for ' + type,
-              category
-            );
-            log.debugc(() => inspection, category);
-            docDef.initCfg.inspectionResultContents = inspection;
-            this.removeInitializationTask();
-          },
-          err => {
-            log.warnc(
-              () =>
-                'No precomputed java document found for ' + type + ': ' + err,
-              category
-            );
-            this.removeInitializationTask();
-          }
-        );
+
+        if ( specification != '' ) {
+          docDef.initCfg.inspectionResultContents = specification;
+        } else {
+          this.addInitializationTask();
+          this.support.requestJavaInspection(connectorId, type).toPromise().then(
+            data => {
+              const inspection: string = data['_body'];
+              log.infoc(
+                () => 'Precomputed java document found for ' + type,
+                category
+              );
+              log.debugc(() => inspection, category);
+              docDef.initCfg.inspectionResultContents = inspection;
+              this.removeInitializationTask();
+            },
+            err => {
+              log.warnc(
+                () =>
+                  'No precomputed java document found for ' + type + ': ' + err,
+                category
```

Comment:
The linter is also configured to complain when indenting mismatches are detected in the same code block, but dunno if this one would get detected. Just for the sake of readability, I'd suggest to align both the string with the `category` variable in the lambda payload. BTW, when chaining two or more strings and variables, ES6 templated strings are the way to go:

```javascript
log.warnc(
  () =>
    `No precomputed java document found for ${type}: ${err}`,
    category
);
```

Reply:
will do next time.  Good tip.